packages/env/lib/commands/restore.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+
+const util = require( 'util' );
+const exec = util.promisify( require( 'child_process' ).exec );
+/**
+ * Internal dependencies
+ */
+const initConfig = require( '../init-config' );
+
+/**
+ * @typedef {import('../wordpress').WPEnvironment} WPEnvironment
+ * @typedef {import('../wordpress').WPEnvironmentSelection} WPEnvironmentSelection
+ */
+
+/**
+ * Takes a snapshot of the development server's database.
+ *
+ * @param {Object}                 options
+ * @param {Object}                 options.spinner     A CLI spinner which indicates progress.
+ * @param {boolean}                options.debug       True if debug mode is enabled.
+ */
+module.exports = async function snapshot( { spinner, debug } ) {
+	const config = await initConfig( { spinner, debug } );
+
+	const snapshotPath = `${ config.configDirectoryPath }/snapshot`;
+	const directoryHash = path.basename( config.workDirectoryPath );
+	const filesVolume = `${ directoryHash }_wordpress`;
+	const dbVolume = `${ directoryHash }_mysql`;
+
+	spinner.text = `Restoring to ${ directoryHash }.`;
```

Comment:
We should probably confirm with the user before resetting their database here. I also think it would be useful to do a quick check that everything is still working correctly at the end, but not exactly sure how.

Reply:
> Really cool, thanks for taking a look! I wonder, does it make more sense to name it `import` and `export`, and then to store the exported data outside of the hidden config folder? This way you could send someone the exported file and then they could import it. Maybe it saves by default to `./wp-env-export.zip`, which could contain both the files and DB data.
> 
> As is, it seems like the script performs best for (e.g.) rolling back your database to a previous snapshot, but I think it would be more flexible to allow for arbitrary import/export locally 🤔
> 
Yes, absolutely.  Passing small snapshots of the app around is my whole goal.  
>
> I also wonder how this would work if you have a different .wp-config.json file locally. For example, maybe you have a different set of plugins or themes, and some different wp-config values.
The dream I see is a .wp-config.json file that is as much a part of a theme's configuration as a package.json file is.  Not necessary, but increasingly common.  And when you "start" that theme project a wp-env environment is started.  Because of a flag in the .wp-config.json file if there is a "save file" in the quick-save folder then load that (load meaning import the state into the volumes). Otherwise make a new game and start up a fresh new instance of WordPress.  
Now I can bounce around to different themes and every time I spin this up, with a single 'npm start' command, I get a WordPress environment that is specifically tailored for development of that theme.

and when shutting down an environment it gives you the option to 'export' the state so that you can pick up where you left off with that theme.

And with wp-env instances sandboxed as they are, with a small port configuration tweak a dev could run multiple themes locally side-by-side (to move content around, compare implementation, etc) and both of them would have their own unique environment specific to the needs of demonstrating and building that theme.

Reply:
> When shutting down an environment it gives you the option to 'export' the state so that you can pick up where you left off with that theme.

Yeah, I guess theoretically you might be able to commit it to git, but unsure how well that would go :P Then all of the files necessary would be downloaded by the same command.

> And with wp-env instances sandboxed as they are, with a small port configuration tweak a dev could run multiple themes locally side-by-side

Totally, that's a good use case.